src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,261 @@
+        internal static DiagnosticDescriptor PropertyPublicRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessagePropertyPublic,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescriptionPropertyPublic,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+        internal static DiagnosticDescriptor FieldRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageField,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescription,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(PropertyNameRule, FieldRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(
+                (compilationStartContext) =>
+                {
+                    INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                    if (jsonConstructorAttributeNamedSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                    compilationStartContext.RegisterOperationBlockStartAction(
+                        (startOperationBlockContext) =>
+                        {
+                            if (startOperationBlockContext.OwningSymbol is IMethodSymbol method
```

Comment:
```suggestion
                            if (context.OwningSymbol is IMethodSymbol method
```

Reply:
I've changed the implementation a bit, most likely it doesn't apply anymore. You may take a look.